CMakeLists.txt

```
@@ -127,9 +127,12 @@ find_package(ZLIB REQUIRED)
 
 if(NOT TARGET CURL::CURL)
   add_library(CURL::CURL INTERFACE IMPORTED)
-  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS)
+  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS OR ENABLE_PYTHON_BINDINGS)
     if(NOT CURL_FOUND)
       find_package(CURL REQUIRED)
+      if (WIN32 AND ENABLE_PYTHON_BINDINGS)
+        link_libraries(${CURL_LIBRARIES})
+      endif()
     endif()
     set_target_properties(CURL::CURL PROPERTIES
       #INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}"
```

maybe this will work work:


```suggestion
  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS)
    if(NOT CURL_FOUND)
      find_package(CURL REQUIRED)
    endif()
    set_target_properties(CURL::CURL PROPERTIES
            INTERFACE_INCLUDE_DIRECTORIES "${CURL_INCLUDE_DIR}"
      INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
    set_property(TARGET CURL::CURL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}")
  else()
    message(STATUS "Using curl from the outside")
  endif()
```

and failing that, maybe could works:

```suggestion
  if(ENABLE_HTTP OR ENABLE_DATA_TOOLS)
    if(NOT CURL_FOUND)
      find_package(CURL REQUIRED)
    endif()
    set_target_properties(CURL::CURL PROPERTIES
            INTERFACE_INCLUDE_DIRECTORIES "${CURL_INCLUDE_DIR}"
      INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
    if(NOT MSVC)
      set_property(TARGET CURL::CURL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}")
      else()
        link_libraries(${CURL_LIBRARIES})
      endif()
    endif()
  else()
    message(STATUS "Using curl from the outside")
  endif()
```

---

Wonderful @kevinkreiser ! and easy.. should've tried to understand the logic here better.. 

Note, the first example errors on
```
[cmake] CMake Error at CMakeLists.txt:137 (set_property):
[cmake]   Property INTERFACE_LINK_LIBRARIES may not contain link-type keyword
[cmake]   "optimized".  The INTERFACE_LINK_LIBRARIES property may contain
[cmake]   configuration-sensitive generator-expressions which may be used to specify
[cmake]   per-configuration rules.
```
Which does sound like a libcurl (or vcpkg?) issue.. didn't find much but same issues for other libs suggest it's the lib's CMake file being the issue. 

Anyways, second option works, not as nice option1, but still a lot better than mine! Fine for you too @kevinkreiser ?